oio/cli/admin/service_check.py

```diff
@@ -140,19 +146,33 @@ def _take_action(self, parsed_args):
                         addr, _ = zookeeper.get(zh.get(), path + '/' + node)
                         registered.add(addr)
                     known = set()
-                    for _, host, port, _ in self.filter_services(self.catalog,
-                                                                 'meta0'):
+
+                    for _, host, port, score in self.filter_services(
+                            self.catalog, 'meta0'):
+                        if score == 0:
+                            yield ('Warn', '%s:%d is down' % (host, port))
                         known.add('%s:%d' % (host, port))
                     self.logger.info("meta0 known=%d zk_registered=%d",
                                      len(known), len(registered))
-                    assert registered == known
+
+                    for item in registered - known:
+                        self.success = False
+                        yield ('Error', '%s not registered but present in zk' %
+                                        item)
+
+                    for item in known - registered:
+                        self.success = False
+                        yield ('Error', '%s registered but not present in zk' %
+                                        item)
```

Comment:
I think "known" and "unknown" are better than "registered" and "not registered".